### PR TITLE
docs: Adding documentation for the DevWorkspace Operator release process

### PR DESCRIPTION
**Pull request by ibuziuk**
Signed-off-by: Ilya Buziuk <ibuziuk@redhat.com>

### What does this PR do?
 Adding documentation for the DevWorkspace Operator release process

### What issues does this PR fix or reference?

N/A

### Is it tested? How?
See preview on https://github.com/ibuziuk/devworkspace-operator/tree/release-docs/docs/release

### PR Checklist

- [ ] E2E tests pass (when PR is ready, comment `/test v8-devworkspace-operator-e2e, v8-che-happy-path` to trigger)
    - [ ] `v8-devworkspace-operator-e2e`: DevWorkspace e2e test
    - [ ] `v8-che-happy-path`: Happy path for verification integration with Che

### CODE REVIEW

File: docs/release/README.md [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+The release process is powered by the 'Release DevWorkspace Operator' [GitHub Action](https://github.com/devfile/devworkspace-operator/actions/workflows/release.yml).
+
+In order to release a new version of the DevWorkspace Operator:
+
+1. Trigger the prerelease by running the action using the workflow from the `main` branch. Provide the version in format `v0.y.z` e.g. `v0.13.0` and set `true` for the prelease field:
+
+![Prerelease](prerelease.png?raw=true "Prerelease")
+
+The action will create the dedicated release branch e.g. `0.13.x` and the prelease commit e.g https://github.com/devfile/devworkspace-operator/commit/4905c6c695e4d4945a3810df9a46a5ecf11d09f1 
+
```
Comment (amisevsk) [inline docs/release/README.md:16]: We could include a step here like 
> 2. If necessary, cherry-pick any additional fixes to the release branch

**Review by amisevsk — APPROVED** [review]
LGTM, thanks!

File: docs/release/README.md (hunk 1 above)
Comment (ibuziuk) [reply docs/release/README.md:16]: added the warning

**Comment by ibuziuk** [comment]
@musienko-maxim looks like prow CI stuck and prevents PR from being merged. Could you please take a look?

**Pull request merged**